static/app/components/modals/inviteMembersModal/inviteRowControlNew.tsx

```diff
@@ -82,6 +89,13 @@ function InviteRowControl({roleDisabledUnallowed, roleOptions}: Props) {
     }
   };
 
+  useEffect(() => {
+    if (isOverMemberLimit) {
+      setRole('billing', 0);
+      setTeams([], 0);
+    }
+  });
+
```

Comment:
I'm wondering if it would be possible to change the role of `defaultInvite` instead? Maybe we could change `defaultInvite()` to take in an optional role parameter and call `defaultInvite("billing")` if the org is over the member limit. That way we could avoid using a `useEffect` here

https://github.com/getsentry/sentry/blob/61f0f3ef082b9f148eceaf950daa1a59ff736a59/static/app/components/modals/inviteMembersModal/useInviteModal.tsx#L21-L27

Reply:
Good question! Yes, we can make adjustments to [defaultInvite()](https://github.com/getsentry/sentry/blob/61f0f3ef082b9f148eceaf950daa1a59ff736a59/static/app/components/modals/inviteMembersModal/useInviteModal.tsx#L21-L27) or the `role` const in [inviteRowControlNew.tsx](https://github.com/getsentry/sentry/blob/453eb93dd62fc10271ac97a336432e142840594c/static/app/components/modals/inviteMembersModal/inviteRowControlNew.tsx#L42) to correctly display the billing role, but we run into a problem when we submit the invite requests. The request json is generated from the state, not the values in the UI elements. I tried a few approaches, but we need to execute both `setRole()` and `setTeams()` in order to set the [state](https://github.com/getsentry/sentry/blob/144d2f0c0539c7a5c65c40bc81cec97a9707047b/static/app/components/modals/inviteMembersModal/useInviteModal.tsx#L229) before executing the POST.

**Without updating state**
```json
{"email":"fails@sentry.io","teams":["hotel7006"],"role":"member"}
```

**With useEffect**
```json
{"email":"works@sentry.io","teams":[],"role":"billing"}
```

I was able to update the useEffect definition, which eliminates unnecessary renders in [453eb93](https://github.com/getsentry/sentry/pull/82379/commits/453eb93dd62fc10271ac97a336432e142840594c).